buffer.c

```diff
@@ -454,25 +371,20 @@ zval *buffer_view_offset_get(buffer_view_object *intern, size_t offset)
 			/* Should never happen */
 			zend_error_noreturn(E_ERROR, "Invalid buffer view type");
 	}
-
-	return retval;
 }
 
 void buffer_view_offset_set(buffer_view_object *intern, long offset, zval *value)
 {
 	if (intern->type == buffer_view_float || intern->type == buffer_view_double) {
-		Z_ADDREF_P(value);
-		convert_to_double_ex(&value);
-
 		if (intern->type == buffer_view_float) {
-			intern->buf.as_float[offset] = Z_DVAL_P(value);
+			intern->buf.as_float[offset] = zval_get_double(value);
 		} else {
-			intern->buf.as_double[offset] = Z_DVAL_P(value);
+			intern->buf.as_double[offset] = zval_get_double(value);
 		}
 
 		zval_ptr_dtor(&value);
 	} else {
-		long lval = get_long_from_zval(value);
+		long lval = zval_get_long(value);
```

Comment:
zend_long

Reply:
Yes. I also changed other variables receiving the output of `zval_get_long` to `zend_long`.